crypto/src/commitment.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use alloc::vec::Vec;
+use core::fmt::Debug;
+
+use utils::{Deserializable, Serializable};
+
+use crate::Hasher;
+
+/// A vector commitment (VC) scheme.
+///
+/// This is a cryptographic primitive allowing one to commit, using a commitment string `com`, to
+/// a vector of values (v_0, ..., v_{n-1}) such that one can later reveal the value at the i-th
+/// position.
+/// This is achieved by providing the value `v_i` together with a proof `proof_i` such that anyone
+/// posessing `com` can be convinced, with high confidence, that the claim is true.
+///
+/// Vector commitment schemes usually have some batching properties in the sense that opening
+/// proofs for a number of `(i, v_i)` can be batched together into one batch opening proof in order
+/// to optimize both the proof size as well as the verification time.
+pub trait VectorCommitment<H: Hasher>: Sized {
```

Comment:
Not for this PR but when we add `Item` associated type, we may consider adding a trait bound on it to build items from lists of field elements - e.g., something like:
```rust
type Item: Serializable + Deserializable + From<&[Self::FieldElement]>;
```
This would also mean that we'd need to add `FieldElement` associated type to the trait.

Reply:
Created issue